Add null check for ParentJoinFieldMapper in ChildrenAggregationBuilder.joinFieldResolveConfig

A user received a NPE. It was reproduced on 6.3.1 with the following stack trace:
```
2019-06-06T13:26:49,963][DEBUG][o.e.a.s.TransportSearchAction] [hamilton] [mi_indice][2], node[ABwM3KQJQdWuKfnzwJ8pxw], [P], s[STARTED], a[id=cRQiXG2DTmijJ2IMjtYq5g]: Failed to execute [SearchRequest{searchType=QUERY_THEN_FETCH, indices=[mi_indice], indicesOptions=IndicesOptions[id=38, ignore_unavailable=false, allow_no_indices=true, expand_wildcards_open=true, expand_wildcards_closed=false, allow_aliases_to_multiple_indices=true, forbid_closed_indices=true, ignore_aliases=false], types=[], routing='null', preference='null', requestCache=null, scroll=null, maxConcurrentShardRequests=15, batchedReduceSize=512, preFilterShardSize=128, allowPartialSearchResults=true, source={"query":{"bool":{"must":[{"bool":{"must":[{"bool":{"should":[{"range":{"posting_date":{"from":"1900-01-01T00:00:00.000Z","to":"2019-05-30T18:25:49.967Z","include_lower":true,"include_upper":true,"boost":1.0}}}],"adjust_pure_negative":true,"boost":1.0}},{"term":{"externaljob_to_child_fields.name":{"value":"externaljob","boost":1.0}}}],"adjust_pure_negative":true,"boost":1.0}}],"adjust_pure_negative":true,"boost":1.0}},"aggregations":{"POSTING_DATES":{"date_range":{"field":"posting_date","ranges":[{"key":"ORA_LT07","from":"2019-05-23T18:25:50.006Z","to":"2019-05-30T18:25:50.006Z"},{"key":"ORA_LT30","from":"2019-04-30T18:25:50.006Z","to":"2019-05-30T18:25:50.006Z"},{"key":"ORA_GT30","from":"1900-01-01T00:00:00.000Z","to":"2019-04-30T18:25:50.006Z"}],"keyed":false}},"LOCATIONS":{"children":{"type":"externaljob_locs"}}}}}] lastShard [true]
org.elasticsearch.transport.RemoteTransportException: [washington][127.0.0.1:9302][indices:data/read/search[phase/query]]
Caused by: java.lang.NullPointerException
	at org.elasticsearch.join.aggregations.ChildrenAggregationBuilder.joinFieldResolveConfig(ChildrenAggregationBuilder.java:122) ~[?:?]
	at org.elasticsearch.join.aggregations.ChildrenAggregationBuilder.resolveConfig(ChildrenAggregationBuilder.java:113) ~[?:?]
	at org.elasticsearch.search.aggregations.support.ValuesSourceAggregationBuilder.doBuild(ValuesSourceAggregationBuilder.java:311) ~[elasticsearch-6.3.1.jar:6.3.1]
	at org.elasticsearch.search.aggregations.support.ValuesSourceAggregationBuilder.doBuild(ValuesSourceAggregationBuilder.java:38) ~[elasticsearch-6.3.1.jar:6.3.1]
	at org.elasticsearch.search.aggregations.AbstractAggregationBuilder.build(AbstractAggregationBuilder.java:139) ~[elasticsearch-6.3.1.jar:6.3.1]
	at org.elasticsearch.search.aggregations.AggregatorFactories$Builder.build(AggregatorFactories.java:329) ~[elasticsearch-6.3.1.jar:6.3.1]
	at org.elasticsearch.search.SearchService.parseSource(SearchService.java:766) ~[elasticsearch-6.3.1.jar:6.3.1]
	at org.elasticsearch.search.SearchService.createContext(SearchService.java:575) ~[elasticsearch-6.3.1.jar:6.3.1]
```

in `ChildrenAggregationBuilder.joinFieldResolveConfig`

```
private void joinFieldResolveConfig(SearchContext context, ValuesSourceConfig<WithOrdinals> config) {
        ParentJoinFieldMapper parentJoinFieldMapper = ParentJoinFieldMapper.getMapper(context.mapperService()); 
        ParentIdFieldMapper parentIdFieldMapper = parentJoinFieldMapper.getParentIdFieldMapper(childType, false);
        if (parentIdFieldMapper != null) {
...
        } else {
            config.unmapped(true);
        }

```
`ParentJoinFieldMapper.getMapper` returns null when there is no parent-join field in the mapping. We already check to see if `parentIdFieldMapper` is null and use an unmapped. Should probably be doing the same for `parentJoinFieldMapper` as well.

--- a/modules/parent-join/src/main/java/org/elasticsearch/join/aggregations/ChildrenAggregationBuilder.java
+++ b/modules/parent-join/src/main/java/org/elasticsearch/join/aggregations/ChildrenAggregationBuilder.java
@@ -112,17 +112,25 @@ public class ChildrenAggregationBuilder extends ValuesSourceAggregationBuilder<C
     @Override
     protected ValuesSourceConfig resolveConfig(QueryShardContext queryShardContext) {
         ValuesSourceConfig config;
+
         ParentJoinFieldMapper parentJoinFieldMapper = ParentJoinFieldMapper.getMapper(queryShardContext.getMapperService());
-        ParentIdFieldMapper parentIdFieldMapper = parentJoinFieldMapper.getParentIdFieldMapper(childType, false);
-        if (parentIdFieldMapper != null) {
-            parentFilter = parentIdFieldMapper.getParentFilter();
-            childFilter = parentIdFieldMapper.getChildFilter(childType);
-            MappedFieldType fieldType = parentIdFieldMapper.fieldType();
-            config = ValuesSourceConfig.resolveFieldOnly(fieldType, queryShardContext);
-        } else {
+        if (parentJoinFieldMapper == null) {
             // Unmapped field case
             config = ValuesSourceConfig.resolveUnmapped(defaultValueSourceType(), queryShardContext);
+            return config;
         }
+
+        ParentIdFieldMapper parentIdFieldMapper = parentJoinFieldMapper.getParentIdFieldMapper(childType, false);
+        if (parentIdFieldMapper == null) {
+            // Unmapped field case
+            config = ValuesSourceConfig.resolveUnmapped(defaultValueSourceType(), queryShardContext);
+            return config;
+        }
+
+        parentFilter = parentIdFieldMapper.getParentFilter();
+        childFilter = parentIdFieldMapper.getChildFilter(childType);
+        MappedFieldType fieldType = parentIdFieldMapper.fieldType();
+        config = ValuesSourceConfig.resolveFieldOnly(fieldType, queryShardContext);
         return config;
     }
 

--- a/modules/parent-join/src/test/resources/rest-api-spec/test/40_unconfigred.yml
+++ b/modules/parent-join/src/test/resources/rest-api-spec/test/40_unconfigred.yml
@@ -1,0 +1,24 @@
+---
+"unconfigured":
+  - do:
+      index:
+        index: test
+        refresh: true
+        body:
+          join:
+            name: question
+          body: <p>I have Windows 2003 server and i bought a new Windows 2008 server...,
+          title: Whats the best way to file transfer my site from server to a newer one?,
+          tags: [windows-server-2003, windows-server-2008, file-transfer]
+
+  - do:
+      search:
+        index: test
+        body:
+          size: 0
+          aggs:
+            to-answers:
+              children:
+                type: answer
+  - match: { hits.total.value: 1 }
+  - match: { aggregations.to-answers.doc_count: 0 }

